### DISCUSSION TITLE
Renaming Starlite

### DISCUSSION THREAD

**Discussion by Goldziher** [litestar-org/litestar#799] [General]
So, a repeated point of contention is the name of the package. While I personally like Starlite, some people find the similarity to Starlette confusing and difficult. 

I'd like us to reach a decision on (A) should we rename the library and the org? And if we agree that we should actually do this, (B) let's agree on a name.

Note: I originally started out with `synapse`, but it was taken in pypi.

**Comment by cofin** [comment]
While I also like the name Starlite, I can understand why there might be confusion and contention.  I've also noticed that it becomes a topic of discussion quite often.  

While there's a lot of effort in renaming/rebranding everything, I think it may be the best thing we can do for recognition/differentiation in the long term.  

I'm in favor of a change, but I don't have any suggestions just yet.

**Reply by provinzkraut** [reply]
I'd also second a renaming. If we're going to do it, we should do it now. The longer we wait, the more cumbersome and problematic it gets. 

As for the new name, I don't have any immediate suggestions, but I'll see if I can't come up with a few ideas!

**Reply by peterschutt** [reply]
Agree, if this is going to be done, it should be done asap.

The project to merge the authentication libs would be a convenient way to encapsulate the rebranding for those.

**Comment by Goldziher** [comment]
Ok, lemme write some ideas to get the discussion going - I'm literally checking pypi 😁 to eliminate, thus far these are free:

* Light, LightAPI
* Botschaft (German for mission)
* Emissary
* Bridge
* Gateway
* Sphere 
* Orbus (Sphere in Latin)
* Astralapi

**Reply by Goldziher** [reply]
My favorite now:

`juncture`

**Reply by provinzkraut** [reply]
Juncture sounds too harsh IMO. I'd prefer a softer name, that sounds friendlier!

And I can't help myself but to read "Astralapi" as "Astra Lapi" (whatever that means) :upside_down_face:

**Reply by peterschutt** [reply]
I have no problems with any of the above however I think it would be nice to keep some sort of astrological theme to the name, just for some continuity/reference to our origins.

Some suggestions:

- gravitas (I was thinking of how they use gravitational slingshots for efficiency in space travel.. gravitas doesn't mean that at all, but the word sounds nice).
- propellant (meh)
- soleil (french for sun, I like this)

Space themed names seem to be very popular for packages so it's hard to find unused ones!

**Reply by old-syniex** [reply]
Tzedek (Hebrew for jupyter also justice)?

**Reply by Goldziher** [reply]
> Tzedek (Hebrew for jupyter also justice)? 

Shabtai

**Reply by Goldziher** [reply]
* Starborn
* Cornet
* Supermoon / newmoon / bluemoon
* DragonAPI
* Neostar

**Reply by Goldziher** [reply]
Also:

* Stellarlight / stellarlite
* Planeta

**Reply by Goldziher** [reply]
Starlane, starway

Ok I'm done for now

**Reply by Goldziher** [reply]
Oh, this one is free

Galaxy

**Reply by peterschutt** [reply]
https://pypi.org/project/galaxy/

**Reply by Goldziher** [reply]
Also galactic

**Reply by Goldziher** [reply]
So, seems some are taken. 

My two favorite -

* Galaxia, which is Greek fir Galaxy
* Astralight or Astralite, which is starlite but using the Greek word for star (astra).

**Reply by provinzkraut** [reply]
- Astraea (Greek mythology, daughter of the goddess of dawn Eos, goddess of justice, innocence purity and precision)
- Astraeus (Greek mythology, father of Astraea)

**Reply by peterschutt** [reply]
Have added to the aggregate list @provinzkraut

**Reply by Goldziher** [reply]
Morningstar

**Reply by Goldziher** [reply]
Stardrive

**Reply by peterschutt** [reply]
morningstar gone (https://pypi.org/project/morningstar/), added stardrive

**Reply by Goldziher** [reply]
Also stardriver and spacecraft

**Reply by provinzkraut** [reply]
- FacileAPI: facile - "Easy" / "simple" in french
- Lumière (`lumiere`): "Light" in french

**Reply by Goldziher** [reply]
Photonic

**Reply by dantium** [reply]
If I may I would like to suggest: **BrightAPI** as in starlight shining **bright**"

**Comment by peterschutt** [comment]
Thoughts on initially supporting the `starlite` namespace within the new named lib? For example, after `pip install new-starlite`, does `from starlite import Starlite` continue to work, or is that an error? And, if we do support it, should it be deprecated with a warning immediately? This would somewhat smooth over the transition downstream.

WRT to the `starlite` package on pypi, we can make a shim package - basically a simple package with the `new-starlite` as a single dependency that proxies through to the underlying lib. If so, would that point to a specific version of `new-starlite`, or always the latest. Importing from the shim lib could also propagate a warning.

To drive the migration, we should seek out public repos that depend on us and make a PR that modernizes and uses the new name. It would be in our best interests to have as many public examples of usage of the library depending on the new name.

**Reply by provinzkraut** [reply]
I think we definitely need to provide a shim package at first, and import names should continue to work with a deprecation warning for now. 

Installing the shim package should do the same thing as installing the new package, except it adds a deprecation warning. I wonder if there isn't a way to elegantly handle this with `pyproject.toml` / poetry, i.e. to release it under one name, with the same code, except for a script executed at install time that raises a deprecation warning. That would definitely be the smoothest transition (for us and users) if we additionally keep things available `from starlite`.

**Reply by peterschutt** [reply]
Should the shim be locked to the first version under the new name, or always pull in the latest? Lets say someone pins to 1.40.0 which is the shim release, it wouldn't really be pinned if the dependency to the new lib wasn't pinned.

Found this: https://github.com/simonw/pypi-rename

**Reply by provinzkraut** [reply]
Neither. I would propose we simply continue to release the shim alongside the new package for a while, and only pin it when we stop releasing it.

**Reply by peterschutt** [reply]
Right, so we would pin the new-name dependency in the shim for each release, and continue to release under both names for a period of time.

E.g., we release New Name 1.40.1, and Starlite 1.40.1 which has a single dependency that is `new-name==1.40.1`.

Then, we make a new feature release, so we release New Name v1.41.0, and Starlite v1.41.0 that has `new-name==1.41.0` as its dependency.. etc etc.

**Comment by Alphasite** [comment]
Have you considered something close to the current name (for continuity and name recognition), but different enough to be its one thing? 

My suggestion is `Starlight` (or `Star Light` for a subtly different take). 

It’s likely just different enough that people won’t mistake it for starlette in my opinion and as a side benifit, it allows you to maintain the current (very nice) logo.

**Reply by Goldziher** [reply]
> Have you considered something close to the current name (for continuity and name recognition), but different enough to be its one thing? 
> 
> My suggestion is `Starlight` (or `Star Light` for a subtly different take). 
> 
> It’s likely just different enough that people won’t mistake it for starlette in my opinion and as a side benifit, it allows you to maintain the current (very nice) logo. 

Starlight is taken in pypi. Was my first go at the starlite name.

**Reply by Alphasite** [reply]
Hmm, `star-light` is available, not the biggest fan of double barrel names, but it’s an option.

**Comment by Goldziher** [comment]
Ok, it seems that the pypi search is not good enough. As @peterschutt pointed out the way to check this is to Google `pypi <name>`.

**Comment by peterschutt** [comment]
Here's an aggregate of all suggestions thus far, filtered for names already taken:

1. Astraea (Greek mythology, daughter of the goddess of dawn Eos, goddess of justice, innocence purity and precision)
2. Astraeus (Greek mythology, father of Astraea)
3. Astralapi (astra is Greek for star)
4. Astralight
5. Astralite
6. Botschaft (German for mission)
7. Emissary
8. FacileAPI
9. Galaxia (Greek for Galaxy)
10. Gateway
11. Gravitas
12. Juncture
13. LightAPI
14. Lumiere
15. Orbus (Sphere in Latin)
16. Propellant
17. Soleil (French for Sun)
18. Spacecraft
19. Star-light
20. Stardrive
21. Stardriver
22. Starlane
23. Starway
24. Tzedek

**Reply by metaperl** [reply]
Is `Bridge` taken? it was [listed earlier](https://github.com/starlite-api/starlite/discussions/799#discussioncomment-4124858)

**Comment by peterschutt** [comment]
OK I've created a poll: https://github.com/starlite-api/starlite/discussions/800

Each member of the org can suggest 2 names, anyone can vote. Votes close in a week from today.

LMK if you think any of the terms should change.

**Reply by provinzkraut** [reply]
Ranked voting would be nice, since that could make more people more happy with the name, but I guess that's a bit different with GH polls? :D

**Reply by peterschutt** [reply]
My initial instinct was for a 3-2-1 style thing, but there didn't seem to be an option for that.

**Reply by provinzkraut** [reply]
Maybe there's a convenient service for this or ranked voting? I'll have a look!
Maybe we should create a poll about how we poll? :upside_down_face:

**Reply by peterschutt** [reply]
maybe, but taking it off GH will reduce participation IMO

**Reply by provinzkraut** [reply]
Hmm. True. An easy solution would be to just create 2 polls, so everyone can get at least 2 votes in?

**Reply by peterschutt** [reply]
I think that would be too confusing as they would be two separate discussions. If the general consensus is that a single vote PP pole is inadequate, I will pull it and we can work out another strategy. I've asked in discord.

**Reply by provinzkraut** [reply]
Oh, I assumed you could have more than one poll in a discussion :cry:

**Reply by peterschutt** [reply]
It's a shame - I was hoping the poll could be created in this discussion, but a poll is a discussion type.

**Reply by metaperl** [reply]
Those suggested names bear no relation to what this product does... `FastAPI` is an attractive yet descriptive name.... hard to type things "Astraea" are not desired at all.

**Comment by peterschutt** [comment]
>  Maybe we should create a poll about how we poll?

You must work in government?

**Comment by infohash** [comment]
I personally like `starlite` name as is. I have a concern that renaming it will put this project into the category of "Yet another competing project that claims to be faster". Think about how many web frameworks in Python do you ever search twice? We want this project to earn the reputation of industrial use like FastAPI became in 2 years and for that, consistency is important.

There is one advantage of keeping similar names and most mobile brands use this strategy. It gives higher search results on the frontpage of  Google for similar things. Just look at how many mobile brands have "Note" series.

If we write about how starlite is/was related to Starlette, the SEO will show that on the front page. I would rather abstain from voting.

**Reply by peterschutt** [reply]
You are certainly free to nominate `starlite` for the poll, it would be good to have it in there TBH.

**Comment by jonadaly** [comment]
@peterschutt  Couple of suggesitions on the theme of starlight:

Superluminal
Luminance
Incandescent

**Reply by peterschutt** [reply]
Thanks @jonadaly! If you had to pick two, which would you drop?

**Reply by jonadaly** [reply]
I think I'd nominate `superluminal`, and maybe `luminant` for the second one

**Comment by adamhoffstein** [comment]
What about pulsar? It's space related? Fast and easy to say.

**Reply by Goldziher** [reply]
https://pypi.org/project/pulsar/

**Reply by metaperl** [reply]
Easy to type should also come into play --- `superluminal`?? are you kidding me.

**Reply by metaperl** [reply]
This module is not related to space or the solar system. I see no reason to seek a name related to space, the stars or the solar system.

**Reply by peterschutt** [reply]
Hey @metaperl, thanks for your input here. We did a poll of contributors and maintainers, and the majority simply wanted to leave the name as "starlite".

**Comment by infohash** [comment]
> some people find the similarity to Starlette confusing and difficult.

Shouldn't we first check whether it's the minority or the majority who gets confused with the name? I won't bother if it's only a subset of users. They will eventually get used to it. Name changing is a huge decision that will impact the popularity growth rate for some time. Unless we have enough analytics that justifies name changing, I don't agree with moving from already a good name to another. I would rather work on recreating its logo.

**Reply by Goldziher** [reply]
sure, you can post to reddit to ask -- or @peterschutt perhaps could?

**Reply by infohash** [reply]
Reddit seems to be irritated about starlite posts. I think we will have to wait for a while to ask there.

**Reply by peterschutt** [reply]
Squeaky wheel gets the oil. The easiest thing is to do nothing, but if we are going to do it, it will only get harder to do as time wears on.

Lets see what others that are paying attention in here have to say first, and we can relax the dates set in the poll.

**Reply by peterschutt** [reply]
OK, I've kicked the dates of the poll back a few days.

**Reply by Goldziher** [reply]
I think we should wait. Yes it's going to be harder the more we wait, but we need more data.

**Comment by metaperl** [comment]
Why wasnt this poll announced on reddit-python? Regular updates are posted there, yet when something as crucial as name choice is on the menu, we are left in the dark.

**Reply by peterschutt** [reply]
At the time of doing this, there was a fair bit of blow-back from some in the reddit community about the frequency of posts about starlite. Also, the discussion never reached an advanced state as general consensus was that we should just leave the name alone. Thanks for your input.

**Comment by i-Ching** [comment]
- The problem of the spelling Starlite is easy to confused with Starnight, Starlight or Starline ....
- In marketing, the name of FastAPI shows a clear meaning: Fast + API.
- Now Starlite is a genuine independent framework. [**strong enough to live without Starlette**](https://youtu.be/S2PWhYAaaFc?t=96). I suggest Starlite to rename **StrongAPI**.

In future, you will compete with FastAPI in tendering for commerical contracts.

**Strong**API: I read your official documentation including SQLAlchemy, Tortorise, Piccolo ORM plugins and covering Mako and Jinja templates.
Your official supports provide more than FastAPI in SQLAlcehmy ORM and Jinja template only.

What's more, enterpise customers request **strong enough api** not fast enough api
(That's why Django still popular 67.7k stars in github, I remember that Django has accumulated python web market share more than 10+ years.) 

Strong**API**: Now you can sing Cher's song https://youtu.be/S2PWhYAaaFc?t=96 to customers but FastAPI cannot live without Starlette.

Finally I state clearly that I am a Starlette fan and comparing Starlette, Starlite, FastAPI, Sanic, aiohttp, quart and tornado.
I pay attention here because Starlite is rising strong.

**Reply by metaperl** [reply]
> In marketing, the name of FastAPI shows a clear meaning: Fast + API.

Yes, that is a very catchy and appealing name which is directly related to functionality instead of what something is built on. Big win right there.

**Comment by gpasq** [comment]
What about lodestone?  It's magnetic and attractive
